sdk/core/azure-core/src/http/log_policy.cpp

```diff
@@ -30,22 +30,63 @@ std::string TruncateIfLengthy(std::string const& s)
   return s.substr(0, BeginLength) + Ellipsis + s.substr(length - EndLength, EndLength);
 }
 
-std::string GetRequestLogMessage(Request const& request)
+std::string GetRequestLogMessage(
+    Azure::Core::Http::LogOptions const& options,
+    Request const& request)
 {
   std::ostringstream log;
 
-  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " "
-      << request.GetUrl().GetAbsoluteUrl();
+  auto const& requestUrl = request.GetUrl();
+
+  std::string url = requestUrl.GetUrlWithoutQuery();
 
-  for (auto header : request.GetHeaders())
   {
-    log << "\n\t" << header.first << " : " << TruncateIfLengthy(header.second);
+    auto separ = '?';
+    for (auto const& qparam : requestUrl.GetQueryParameters())
+    {
+      url += separ + qparam.first + '=';
+
+      if (options.AllowedHttpQueryParameters.find(qparam.first)
+          != options.AllowedHttpQueryParameters.end())
+      {
+        url += TruncateIfLengthy(qparam.second);
+      }
+      else
+      {
+        url += "[hidden]";
+      }
+
+      separ = '&';
+    }
+  }
+
+  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " " << url;
+
+  for (auto const& header : request.GetHeaders())
+  {
+    log << "\n\t" << header.first;
+
+    if (header.second.empty())
+    {
+      log << " [empty]";
+    }
+    else if (
+        options.AllowedHttpRequestHeaders.find(header.first)
+        != options.AllowedHttpRequestHeaders.end())
+    {
+      log << " : " << TruncateIfLengthy(header.second);
+    }
+    else
+    {
+      log << " [hidden]";
+    }
```

Comment:
Nit:  This code is duplicated 3 times, 
- Request -> query
- Request -> headers
- Response -> headers

Consider creating a helper method which takes the map, and the allowed values set and updates the ofstream.
Then if we want to change the `[hidden]` value or anything in the logic it would be just one place

Reply:
Thank you, I will try to unify.